lib/src/utils/typing_utils.dart

```diff
@@ -0,0 +1,7 @@
+typedef TypeComparisonFunction = bool Function<T1, T2>();
+
+class TypingUtils {
```

Comment:
```suggestion
class TypeUtils {
```
Does it make sense?

Reply:
Oops, didn't realise the ambiguous meaning of the word `typing` :)